File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    p = []
    for idx in range(len(words)):
        if p and p[-1][-1] != words[idx][0]:
            return [idx%n+1, idx//n+1]
        if words[idx] in p:
            return [idx%n+1, idx//n+1]
        p.append(words[idx])
    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    const dictionary = {}
    const length = words.length;
    let count = 0;
    
    
    for(let i = 0; i < length; i++) {
        if(words[i - 1] !== undefined && (words[i - 1].at(-1) !== words[i].at(0))) break
        if(dictionary[words[i]]) break
        dictionary[words[i]] = true
        count++
    }
    
    
    if(words.length === count) return [0, 0]
    
    return [(count % n) + 1, parseInt(count / n) + 1]
}